### DISCUSSION TITLE
Connecting to server from local network

### DISCUSSION THREAD

**Discussion by wendallsan** [almarklein/timetagger#167] [Q&A]
Hi, I'm evaluating this as a tool for my household and have stood up a server which I can reach on the system running it at localhost:80, but not from other systems on the local network.  I'm running the server on a Windows 11 system for the moment, and have open port 80 up in the Windows Firewall, so I THINK it should be open.  Is there anything I can do to make this available on my local network?

**Comment by almarklein** [comment]
A server hosting at "localhost:80" (which is an alias to "127.0.0.1:80") is only reachable by clients on the same machine. It actually skips a few networking layers.

To make a server public, use "0.0.0.0:80" instead.

*edit:* the default login procedure checks that the incoming connection is localhost, so you may want to adjus that. Either remove the check (if you can safely assume that the server is really local to the network), or replace it with an alternative authentication procedure. For the latter I don't have concrete examples, but keep an eye on #145.

**Reply by wendallsan** [reply]
Hi and thanks for the response!  Let me elaborate:  I'm not able to reach the server on my local network when trying to reach it by the server's local IP address rather than trying to reach it by typing 'localhost' into another system on the network.  The server's local IP is 192.168.0.22 and that is what I'm using to try to access the server from other systems on the local network.  Requests don't seem to get any response and the browser eventually returns a timeout error.  

I see you're saying that I should use 0.0.0.0:80 instead of localhost, and I see this value in the config.py file, but changing it to my IP plus port 80 didn't seem to help anything (and also broke my ability to access the server via localhost on the system running the server).  I ran a search for 'localhost' in the project and see it is pretty heavily dispersed throughout the project, can you identify where this check is occurring and I'll try to take a look at it from there?

**Reply by almarklein** [reply]
You should set the bind config value to literally `0.0.0.0:80' (which is actually the default). This IP means that it will accept any incoming connection - you can type "localhost" in a browser on the same machine. Or (in your case) "192.168.0.22" from another machine on the local network.

If with this setting you still can't access the server from the local network, there is likely another problem. I'd double check that the IP is correct by pinging the server, and running `ipconfig` on the server. And double check the firewall, e.g. disable it completely for a moment.

**Reply by Tagirijus** [reply]
Sorry for maybe re-opening this discussion, but: I was able to run the server on a machine and connect with my mobile. Yet if I want to install the PWA via Firefox on Android, it's not possible, due to missing SSL, I guess? I mean the link only works with http, yet not with https. Am I missing something?

**Reply by almarklein** [reply]
It [looks like](https://developer.chrome.com/docs/lighthouse/pwa/is-on-https) *A page can't qualify as a Progressive Web App (PWA) if it doesn't run on HTTPS*.

**Reply by Tagirijus** [reply]
Thanks for your reply! In the meantime I am trying to get my own webserver, which is hosted somewhere else instead. Getting HTTPS then, I think. Looking forward to use TimeTagger on a real own server then!
